var/spack/repos/builtin/packages/qscintilla/package.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+
+
+class Qscintilla(QMakePackage):
+    """
+    QScintilla is a port to Qt of Neil Hodgson's Scintilla C++ editor control.
+    """
+
+    homepage = "https://www.riverbankcomputing.com/software/qscintilla/intro"
+    url      = "https://www.riverbankcomputing.com/static/Downloads/QScintilla/2.11.2/QScintilla_gpl-2.11.2.tar.gz"
+
+    version('2.11.2', sha256='029bdc476a069fda2cea3cd937ba19cc7fa614fb90578caef98ed703b658f4a1')
+    # Newer versions of Qscintilla won't build, so prefer the following version
+    version('2.10.2', sha256='14b31d20717eed95ea9bea4cd16e5e1b72cee7ebac647cba878e0f6db6a65ed0', preferred=True)
+
+    variant('designer', default=False, description="Enable pluging for Qt-Designer")
+    # No 'python' variant, since Python bindings will be
+    # built by PyQt5+qsci instead
+
+    depends_on('qt')
+
+    @run_before('qmake')
+    def chdir(self):
+        os.chdir(str(self.stage.source_path) + '/Qt4Qt5')
```

Comment:
You should probably update the `QMakePackage` base class to support a `qmake_directory` or something like that, similar to `CMakePackage` and `AutotoolsPackage`.